java/src/main/java/com/plaid/quickstart/resources/TransactionsResource.java

```diff
@@ -0,0 +1,58 @@
+package com.plaid.quickstart.resources;
+
+import java.io.IOException;
+import java.util.Date;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import com.plaid.client.PlaidClient;
+import com.plaid.client.request.TransactionsGetRequest;
+import com.plaid.client.response.ErrorResponse;
+import com.plaid.client.response.ItemStatus;
+import com.plaid.client.response.TransactionsGetResponse;
+import com.plaid.quickstart.QuickstartApplication;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import retrofit2.Response;
+
+@Path("/transactions")
+@Produces(MediaType.APPLICATION_JSON)
+public class TransactionsResource {
+  private PlaidClient plaidClient;
+
+  public TransactionsResource(PlaidClient plaidClient) {
+    this.plaidClient = plaidClient;
+  }
+
+  @POST
+  public TransactionsGetResponse getTransactions() throws IOException {
+    String accessToken = QuickstartApplication.accessToken;
+    Date startDate = new Date(System.currentTimeMillis() - 86400000L * 100);
+    Date endDate = new Date();
+
+    TransactionsGetRequest request =
+      new TransactionsGetRequest(accessToken, startDate, endDate)
+        .withCount(100);
+
+    Response<TransactionsGetResponse> response = null;
+    for (int i = 0; i < 5; i++) {
+      response = plaidClient.service().transactionsGet(request).execute();
+      if (response.isSuccessful()) {
+        break;
+      } else {
+        try {
+          ErrorResponse errorResponse = plaidClient.parseError(response);
```

Comment:
Are we handling `PRODUCT_NOT_READY` errors correctly? The other Quickstart implementations make sure that's surfaced.

Reply:
I believe that is handled in the view file - we return the `response.body()` at the bottom of this POST and should propagate down to the client, but have not been able to test this logic